buildSrc/src/main/groovy/org/gradle/modules/PatchExternalModules.groovy

```diff
@@ -28,40 +29,46 @@ import org.gradle.api.tasks.*
 class PatchExternalModules extends DefaultTask {
 
     @Internal
-    Configuration externalModules
+    Configuration modulesToPatch
 
     @Input
-    Set<String> getExternalModuleNames() {
-        externalModules.dependencies*.name as Set
+    Set<String> getNamesOfModulesToPatch() {
+        modulesToPatch.dependencies*.name as Set
     }
 
-    @Classpath
-    Configuration externalModulesRuntime
+    @Internal
+    Configuration allModules
+
+    @Internal
+    Configuration coreModules
 
     @Classpath
-    Configuration coreRuntime
+    FileCollection getExternalModules() {
+        allModules - coreModules
+    }
 
     @OutputDirectory
     File destination
 
     PatchExternalModules() {
         description = 'Patches the classpath manifests of external modules such as gradle-script-kotlin to match the Gradle runtime configuration.'
+        dependsOn { modulesToPatch }
     }
 
     @TaskAction
     void patch() {
         ((ProjectInternal) project).sync { CopySpec copySpec ->
-            copySpec.from(externalModulesRuntime - coreRuntime)
+            copySpec.from(externalModules)
             copySpec.into(destination)
         }
 
         def rootProject = project.rootProject as ProjectInternal
 
-        new ClasspathManifestPatcher(rootProject, temporaryDir, externalModulesRuntime, externalModuleNames)
+        new ClasspathManifestPatcher(rootProject, temporaryDir, allModules, namesOfModulesToPatch)
                 .writePatchedFilesTo(destination)
 
         // TODO: Should this be configurable?
-        new ExcludeEntryPatcher(rootProject, temporaryDir, externalModulesRuntime, "kotlin-compiler-embeddable")
+        new ExcludeEntryPatcher(rootProject, temporaryDir, allModules, "kotlin-compiler-embeddable")
```

Comment:
If `kotlin-complier-embeddable` ends up on the Gradle core dependency, then it would be patched, too? Currently, this code assumes that is is only part of the `externalModules`, right?